### PR TITLE
Fixed hyperloglog type assertion check.

### PR DESCRIPTION
**Pull request by AshwinKul28**
The Dice server was breaking with the wrong `pfadd` command. 
This has been caught using TCL Redis tests.

Issue:
Screenshot 2024-09-12 at 1 16 55 AM

Fix:
Screenshot 2024-09-12 at 1 34 42 AM

### CODE REVIEW

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Can we add a small unit test in eval_test.go for this?

File: internal/eval/eval.go [hunk 1]
```diff
@@ -2570,7 +2570,10 @@ func evalPFADD(args []string, store *dstore.Store) []byte {
 		return clientio.Encode(1, false)
 	}
 
-	existingHll := obj.Value.(*hyperloglog.Sketch)
+	existingHll, ok := obj.Value.(*hyperloglog.Sketch)
+	if !ok {
+		return diceerrors.NewErrWithMessage("*WRONGTYPE*")
```
Comment (lucifercr07) [inline internal/eval/eval.go:2575]: We shall change the error as below to maintain parity with Redis.
`WrongTypeHllErr    = "-WRONGTYPE Key is not a valid HyperLogLog string value."`

```
if !ok {
  return diceerrors.NewErrWithMessage(diceerrors.WrongTypeErr)
}
```

Output from Redis:
```
127.0.0.1:6379> pfadd hll a b c
(integer) 1
127.0.0.1:6379> setrange hll 0 "0123"
(integer) 27
127.0.0.1:6379> pfcount hll
(error) WRONGTYPE Key is not a valid HyperLogLog string value.
```

**Review by lucifercr07 — CHANGES_REQUESTED** [review]

File: internal/eval/eval.go (hunk 1 above)
Comment (AshwinKul28) [reply internal/eval/eval.go]: Done. thanks

**Comment by AshwinKul28** [comment]
Added a unit test case.

File: internal/eval/eval_test.go [hunk 2]
```diff
@@ -1109,6 +1121,9 @@ func runEvalTests(t *testing.T, tests map[string]evalTestCase, evalFunc func([]s
 			}
 
 			output := evalFunc(tc.input, store)
+
+			fmt.Println(string(output))
```
Comment (lucifercr07) [inline internal/eval/eval_test.go:1125]: nit: This can be removed.

Comment (AshwinKul28) [reply internal/eval/eval_test.go]: Ahh sorry, bad miss. thanks

**Review by JyotinderSingh — APPROVED** [review]

**Pull request merged**